components/blobserve/pkg/blobserve/blobserve.go

```diff
@@ -75,7 +75,7 @@ func NewServer(cfg Config, resolver ResolverProvider) (*Server, error) {
 // Serve serves the registry on the given port
 func (reg *Server) Serve() error {
 	r := mux.NewRouter()
-	r.PathPrefix(`/{repo:[a-zA-Z0-9\/\-\.]+}:{tag:[a-z0-9][a-z0-9-\.]+}`).MatcherFunc(isNoWebsocketRequest).HandlerFunc(reg.serve)
+	r.PathPrefix(`/{repo:[a-zA-Z0-9\/\-\.\:]+}:{tag:[a-z0-9][a-z0-9-\.]+}`).MatcherFunc(isNoWebsocketRequest).HandlerFunc(reg.serve)
```

Comment:
On first glance I'm not sure if this regex still works: Before `:` was a "hard" separator. Is it guaranteed that we still split repo and tag? [Here](https://github.com/distribution/distribution/blob/main/reference/regexp.go#L72) it looks as if the colon is the anchor between repo and tag...

Reply:
Ah, just found the domain/port separator here](https://github.com/distribution/distribution/blob/main/reference/regexp.go#L38). The difference is that it can only appear once (optionally); instead of repeatedly (which would eat the second one greedily).

Reply:
You can test the implementation with this code snipped:

```go
package main

import (
	"fmt"
	"log"
	"net/http"
	"github.com/gorilla/mux"
)

func main() {
	r := mux.NewRouter()
	r.PathPrefix(`/{repo:[a-zA-Z0-9\/\-\.\:]+}:{tag:[a-z0-9][a-z0-9-\.]+}`).HandlerFunc(serve)
	http.Handle("/", r)
	fmt.Println("Starting up on 8080")
	log.Fatal(http.ListenAndServe(":8080", nil))
}

func serve(w http.ResponseWriter, req *http.Request) {
	vars := mux.Vars(req)
	repo, tag := vars["repo"], vars["tag"]
	fmt.Fprintf(w, "repo: %s, tag: %s", repo, tag)
}
```

When I open https://8080-cyan-horse-0w3rwvnh.ws-eu03.gitpod.io/alpine:v.123 I get:
```
repo: alpine, tag: v.123
```
With https://8080-cyan-horse-0w3rwvnh.ws-eu03.gitpod.io/example.repo.com:9000/alpine:v.123 I get:
```
repo: example.repo.com:9000/alpine, tag: v.123
```

However, an improvement would be to don't split repo and tag with the PathPrefix regex but using github.com/docker/distribution/reference to do the split for us.

Reply:
Ah, got it. It works because in practice we always have a tag.

> However, an improvement would be to don't split repo and tag with the PathPrefix regex but using github.com/docker/distribution/reference to do the split for us.

Absolutely.